presto-main/src/test/java/io/prestosql/execution/TestSqlTaskExecution.java

```diff
@@ -278,7 +282,13 @@ public void testSimple(PipelineExecutionStrategy executionStrategy)
             }
 
             outputBufferConsumer.abort(); // complete the task by calling abort on it
-            TaskState taskState = taskStateMachine.getStateChange(TaskState.RUNNING).get(10, SECONDS);
```

Comment:
I think we can simplify it by:
```
TaskState taskState = taskStateMachine.getStateChange(taskStateMachine.getState()).get(10, SECONDS);
```
not that if `getState` returns `FINISHED`, future will complete immediately.

Reply:
I wanted to do that itself - but the `getStateChange` method throws an exception if the `getState` returns `FINISHED`.
Also, if the state is running, the function might return if the task transitions to `FLUSHING`. And we'd have to run `getStateChange` again.